crates/red_knot_python_semantic/src/semantic_index/ast_ids.rs

```diff
@@ -34,7 +34,13 @@ pub(crate) struct AstIds {
 
 impl AstIds {
     fn expression_id(&self, key: impl Into<ExpressionNodeKey>) -> ScopedExpressionId {
-        self.expressions_map[&key.into()]
+        let key = &key.into();
+        match self.expressions_map.get(key) {
+            Some(result) => *result,
+            None => {
+                panic!("Could not find expression ID for {key:?}");
```

Comment:
This is very helpful because debug output on `ExpressionNodeKey` includes the range, which is usually enough in context to identify what text fragment is causing issues.

I put this in under the idea that this doesn't cost much but I might be wrong